### PR TITLE
Implementation of translate-text function in python

### LINKED ISSUE #97: Implement translateText() Appwrite Function ⚡
# Introduction

Appwrite Functions allow you to extend and customize your Appwrite server functionality by executing your custom code 🤩 You can learn more at our official [Appwrite Functions docs](https://appwrite.io/docs/functions).

Your task is to implement `translateText` function. You can look at the existing [Appwrite Functions demo](https://github.com/appwrite/demos-for-functions) in the coding language you prefer to see how it works.

This function should take `text`, `sourceLanguage` and `destinationLanguage` as an input, use [Google Translate API](https://cloud.google.com/translate) to translate it and output the translated version.

> Please make sure to store all configurable parameters as environment variables, so others can start using your script without having to edit it. If some data is different for each execution, make sure to use `APPWRITE_FUNCTION_EVENT_DATA` or `APPWRITE_FUNCTION_DATA` environment variable.

### Tasks summary:

- [ ] Fork & clone this repository
- [ ] Create `feat-implement-translate-text-X` branch where `X` is programming language you used
- [ ] Write code for the function
- [ ] Test the function using Appwrite
- [ ] Write a `README.md` explaining what function does, how to use and customize it. Here is [an example](https://github.com/appwrite/demos-for-functions/tree/master/nodejs/welcome-email).
- [ ] Submit pull request

If you need any help, reach out to us on our [Discord server](https://discord.gg/GSeTUeA).

Are you ready to work on this issue? 🤔 Let us know, and we will assign it to you 😊 There are 9 languages in which you can implement this function. When requesting to be assigned, please mention which language would you like to use.

### Assignees table

✅ = Done - This language has already been finished
⚒ = In progress - Someone is already working on this language
❌ = Unassigned yet - This language can be assigned to you

| Function / Language | Deno | NodeJS | PHP | Python | Ruby | Dart | .NET | Java | Kotlin |
| ------------------- | ---- | ------ | --- | ------ | ---- | ---- | ---- | ---- | ------ |
| translateText       | ❌   | ❌     | ❌  | ❌     | ❌   | ❌   | ❌   | ❌   | ❌     |

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by avats-dev**
Fixes [1917](https://github.com/appwrite/appwrite/issues/1917)

Created translate-text folder in python directory with `main.py`, `requirements.txt` and `README.md`.

I have used [this translate package](https://github.com/terryyin/translate-python) for implementing this as it doesn't requires any credentials and is compatible with many providers. Other major API providers such as Google Translate, Microsoft Translate and IBM Watson require instances of services on their cloud platform which is a tedious task and to again authenticate them for using with appwrite console. Because of this, I decided to go for this package instead. 

This function can translate a paragraph in one go and doesn't requires source language explicitly, however target language is required otherwise it will fail.

---

## Local testing of function

> Output

>Output

> Output

### CODE REVIEW

File: python/translateText/main.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import six
+from google.cloud import translate_v2 as translate
+
+
+def translateText(text, sourceLanguage, destinationLanguage):
```
Comment (Meldiron) [inline python/translateText/main.py:5]: We need to execute the function and print the response. To get these dynamic input, we can use environment variable `APPWRITE_FUNCTION_DATA` (it's content can be a json that you can then parse and use. please document expected input in readme).

File: python/translateText/main.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    translateClient = translate.Client()
```
Comment (Meldiron) [inline python/translateText/main.py:6]: Yeah, I would be scared about not using some Google Cloud KEY or something.. Can we figure this one out? What does their permission keys file store? Is that a JSON? Can we just put it into the environment variable?

File: python/translateText/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Translate text from one language to another
+A sample python Cloud Function for translating text from one language to another using google translate API.
+
+## 📝 Environment Variables
+Go to Settings tab of your Cloud Function. Add the following environment variables.
+
+* **GOOGLE_APPLICATION_CREDENTIALS** - Your google json API key path
```
Comment (Meldiron) [inline python/translateText/README.md:7]: Where do we use this variable? We surly need to authentificate.

File: python/translateText/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+$ cd ..
+$ tar -zcvf code.tar.gz translateText
+```
+
+* Navigate to the Overview Tab of your Cloud Function > Deploy Tag
+* Input the command that will run your function (in this case `python main.py`) as your entrypoint command
+* Upload your tarfile
+* Click 'Activate'
+
+## Trigger
```
Comment (Meldiron) [inline python/translateText/README.md:39]: Let's say that trigger is manual so either manually from console or using `executeFunction` from SDK

**Review by Meldiron — CHANGES_REQUESTED** [review]
Great PR! 🤯 Left some comments during the review, please check them out.

I reviewed your other PR first, please check that one out first, some points might apply to this PR too.

**Comment by Meldiron** [comment]
Also please rename folder from `translateText` to `translate-text` to be consistent with the rest of the repo

**Comment by avats-dev** [comment]
@Meldiron I have updated the PR and tested it locally and also have attached screenshots for the same. Do let me know if you need any more tweaks. :tada:

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+is%3Aissue+label%3Ahacktoberfest+created%3A%3E2021-01-01+is%3Aopen&type=issues) and help make Appwrite better :)

**Pull request merged**